doc/target.rst

```diff
@@ -267,6 +267,10 @@ Target
         pattern instead of being take as-is. If the pattern has multiple
         matches, ``dest`` must be a folder (or will be created as such if it
         does not exists yet).
+   :param via_temp: If ``True``, copy the file first to a temporary location on
```

Comment:
IIUC the `via_temp` argument is essentially doing the same behavior as the `as_root` parameter was doing implicitly. If so I wonder if there any usecase that would benefit by allowing the temporary file approach to be used without root as well? 
However either way it might make sense to duplicate whatever we end up with for the `push` abi for consistency as we have the implementation anyway.

Reply:
The motivation is to be able to download large sysfs files at a reasonable speed, so it's an alternative workaround to: https://github.com/ARM-software/devlib/pull/562

Currently, pulling 3.5MB from sysfs takes 36 seconds. With via_temp=True, it takes 2s.

I don't know if there is any use for push via_temp though, as it's really just an ugly workaround. If in the future via_temp becomes a burden to maintain and OpenSSH fixes its handling (or paramiko fixes its prefetching logic), we could make via_temp a no-op.